### PR TITLE
Give pause state tracking from pieces to the board

### PR DESCRIPTION
**Pull request by en-chang**
I sent some messages to you and Honjae about how we want to make the paused game to look. I did the quicker easier option first just to give us something to work with but yeah I'm pretty sure I didn't follow proper coding convention for some things so just let me know. I pretty much just made it the same as the overlay for finishing the game except not as dark and more blurry so people can't cheat.
- Made it to where pieces aren't tracking the paused state
- Add an overlay when paused
paused game

### CODE REVIEW

File: src/components/GameBoard/GameBoard.tsx [hunk 1]
```diff
@@ -79,6 +83,9 @@ const GameBoard: FC<GameBoardProps> = ({
           highlightedNumber={highlightedNumber}
           pencilMarkBoard={pencilMarkBoard}
         />
+        {isPaused === true && (
```
Comment (bkchu) [inline src/components/GameBoard/GameBoard.tsx]: Can simplify to
```jsx
{isPaused && (
```

File: src/components/GameBoard/Piece/Piece.tsx [hunk 2]
```diff
@@ -65,20 +61,16 @@ const Piece: FC<PieceProps> = ({
             isActive={isActive}
           ></PencilGrid>
         )}
-        {number === 0 ? "" : isPaused ? "" : `${number}`}
+        {number === 0 ? "" : `${number}`}
```
Comment (bkchu) [inline src/components/GameBoard/Piece/Piece.tsx]:
```suggestion
        {number === 0 ? "" : number}
```

File: src/components/GameBoard/Piece/Piece.tsx [hunk 3]
```diff
@@ -65,20 +61,16 @@ const Piece: FC<PieceProps> = ({
...
-})
-
 const mapDispatchToProps = (dispatch: Dispatch) => ({
   onPieceClick: index => dispatch(createBoardSelectPieceAction(index)),
   setHighlightedNumber: number =>
     dispatch(createBoardSetHighlightedNumber(number)),
 })
 
-export default connect(mapStateToProps, mapDispatchToProps)(Piece)
+export default connect(mapDispatchToProps)(Piece)
```
Comment (bkchu) [inline src/components/GameBoard/Piece/Piece.tsx]:
```suggestion
export default connect(null, mapDispatchToProps)(Piece)
```

Comment (bkchu) [reply src/components/GameBoard/Piece/Piece.tsx]: You still need the first parameter, since the second parameter is for the `mapDispatchToProps`.

**Review by bkchu — CHANGES_REQUESTED** [review]
Small changes, but otherwise looks solid.

**Comment by en-chang** [comment]
I just made the changes. Do I have to or should I resolve each conversation when you make comments like that? Also I'm going to try to make it appear the same as what we had before, a blank board instead of an overlay, but I don't think I'll be able to do it but I'll try.

**Comment by en-chang** [comment]
I got the board to be like how it was before but I kept the pause text overlay just cause. Honestly I'm satisfied with this and everything else is just UI preferences. I think I'd like for the overlay to be a play button but we can work on that later.
blank board

**Pull request closed without merging**